core/net/src/hybrid/client.rs

```diff
@@ -57,7 +57,7 @@ impl Actor for ClientActor {
         let addr = ctx.address();
         ctx.wait(
             async move {
-                ADDRESS.write().await.replace(addr);
+                ADDRESS.write().unwrap().replace(addr);
```

Comment:
Just moved it to [ClientActor::init](https://github.com/golemfactory/yagna/pull/2056/commits/0571926adaf850f11d637bd409b2d26b0a599a8f#diff-dd6d63f1d0c80bcb08c52a922fe7ee3b6b68ff68fc1a6ae3364b9f3f8925322bR49) so it will be called whenever [Services::gsb](https://github.com/golemfactory/yagna/pull/2056/commits/0571926adaf850f11d637bd409b2d26b0a599a8f#diff-ff490246b403bbf6555dcfcf6cb38d31758d138d6fadbe6cd529de60d4374a99R539) is, and to not mix order there is an ugly check+panic before calling [Services::rest](https://github.com/golemfactory/yagna/pull/2056/commits/0571926adaf850f11d637bd409b2d26b0a599a8f#diff-ff490246b403bbf6555dcfcf6cb38d31758d138d6fadbe6cd529de60d4374a99R562). (calling Services::gsb mut be called before Services::rest to initialize ClientProxy)

Reply:
I moved back to `tokio::sync::RwLock` because I will call ClientProxy in REST handler which is already async, so `async move` can stay.